### PR TITLE
Amjith/background completion

### PR DESCRIPTION
**Pull request by amjith**
Reviewer: @tsroten

The completion data-structures are populated at the startup time and then refreshed every time we ALTER, DROP, CREATE or change the database. But this completion refresh happens synchronously which could tie up the REPL for a while if the database is huge (with hundreds of thousands of tables). So instead of refreshing the completions synchronously this PR will make it asynchronous by kicking off a background thread. 

This is a copy of the feature that was implemented in pgcli a little while back. 

See this PR for discussion: https://github.com/dbcli/pgcli/pull/345

### CODE REVIEW

**Comment by tsroten** [comment]
@amjith This looks like a great feature! Other than my comment in the code, it looks good to me.

File: mycli/main.py [hunk 1]
```diff
@@ -475,7 +484,7 @@ def prompt_tokens(cli):
 
                 # Refresh the table names and column names if necessary.
                 if need_completion_refresh(document.text):
-                    self.refresh_dynamic_completions()
+                    self.refresh_completions(reset=need_completion_reset(document.text))
```
Comment (tsroten) [inline mycli/main.py]: Should you check if the command was successful before starting a thread to refresh completions? If the create/alter/update statement wasn't successful, wouldn't that be a waste of resources, especially if the database is large?

Comment (amjith) [inline mycli/main.py]: You're right I didn't think about that. I'll see what I can do.

**Comment by amjith** [comment]
I've updated the PR to do the completion refresh only if the command was a success.

**Pull request merged**